File: leetcode_solutions/0169-majority-element/solution.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        nums.sort()
        dict = {}
        times = len(nums)//2
        for value in nums:
            if value not in dict:
                dict[value] = 1
            else:
                dict[value] += 1
            if dict[value] > times:
                return value
        return nums[times]

File: leetcode_solutions/0169-majority-element/solution.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        
        int times = nums.size()/ 2;
        map<int, int> m;
        for (auto &n : nums){
            if (m.find(n) == m.end()){
                m[n] = 1;
            }else{
                m[n] += 1;
            }
            if (m[n] > times){
                return n;
            }
        }
        
        return m[times];
    }
};
